### PR TITLE
Updating the supported platforms page

### PR DESCRIPTION
**Pull request by sfshaza2**
Introduce a new meaning of "Supported", drop "Best effort", and introduce a new "CI-tested" category that mostly has the old meaning of "Supported".

Resolves https://github.com/flutter/website/issues/11206

### CODE REVIEW

File: src/_data/platforms.yml [hunk 1]
```diff
@@ -1,50 +1,50 @@
...
-  besteffort: '12 to 16, 18'
+  supported: '12 to 18'
+  ci-tested: '16, 17'
   unsupported: '11 and earlier'
 - platform: 'macOS'
   chipsets: 'x64, Arm64'
-  supported: '<abbr title="Ventura">13</abbr>, <abbr title="Sonoma">14</abbr>'
-  besteffort: '<abbr title="Mojave">10.14</abbr> to <abbr title="Monterey">12</abbr>, <abbr title="Sequoia">15</abbr>'
-  unsupported: '<abbr title="High Sierra">10.13</abbr> and earlier'
+  supported: '10.14 to 10.15'
```
Comment (tvolkert) [inline src/_data/platforms.yml:13]: Mojave (10.14) to Sequoia (15)

File: src/_data/platforms.yml [hunk 2]
```diff
@@ -1,50 +1,50 @@
...
+  ci-tested: '12'
+  unsupported: '10.13 and earlier'
 - platform: 'Windows'
   chipsets: 'x64, Arm64'
   supported: '10'
```
Comment (tvolkert) [inline src/_data/platforms.yml:18]: 10, 11

File: src/_data/platforms.yml [hunk 3]
```diff
@@ -1,50 +1,50 @@
...
-  besteffort: '11'
+  ci-tested:
```
Comment (tvolkert) [inline src/_data/platforms.yml:19]: 10

File: src/_data/platforms.yml [hunk 4]
```diff
@@ -1,50 +1,50 @@
...
+  unsupported: '10.13 and earlier'
 - platform: 'Windows'
   chipsets: 'x64, Arm64'
   supported: '10'
-  besteffort: '11'
+  ci-tested: 
   unsupported: '8 and earlier'
 - platform: 'Debian (Linux)'
   chipsets: 'x64, Arm64'
   supported: '11, 12'
```
Comment (tvolkert) [inline src/_data/platforms.yml:23]: 10, 11, 12

File: src/_data/platforms.yml [hunk 5]
```diff
@@ -1,50 +1,50 @@
...
-  besteffort: '10'
+  ci-tested:
```
Comment (tvolkert) [inline src/_data/platforms.yml:24]: 11, 12

File: src/_data/platforms.yml [hunk 6]
```diff
@@ -1,50 +1,50 @@
...
   chipsets: 'x64, Arm64'
   supported: '11, 12'
-  besteffort: '10'
+  ci-tested: 
   unsupported: '9 and earlier'
 - platform: 'Ubuntu (Linux)'
   chipsets: 'x64, Arm64'
   supported: '20.04 LTS, 22.04 LTS'
-  besteffort: '24.04 LTS'
+  ci-tested:
```
Comment (tvolkert) [inline src/_data/platforms.yml:29]: 22.04 LTS

File: src/_data/platforms.yml [hunk 7]
```diff
@@ -1,50 +1,50 @@
...
+  supported: '12 to 18'
+  ci-tested: '16, 17'
   unsupported: '11 and earlier'
 - platform: 'macOS'
   chipsets: 'x64, Arm64'
-  supported: '<abbr title="Ventura">13</abbr>, <abbr title="Sonoma">14</abbr>'
-  besteffort: '<abbr title="Mojave">10.14</abbr> to <abbr title="Monterey">12</abbr>, <abbr title="Sequoia">15</abbr>'
-  unsupported: '<abbr title="High Sierra">10.13</abbr> and earlier'
+  supported: '10.14 to 10.15'
+  ci-tested: '12'
```
Comment (cbracken) [inline src/_data/platforms.yml:14]: I know @vashworth started the update of our bots to macOS Sonoma (14) a while back, but @jmagman do you know what the breakdown currently looks like off the top of your head? If not, I'll dig it up from the CI config.

File: src/_data/platforms.yml (hunk 1 above)
Comment (cbracken) [reply src/_data/platforms.yml:13]: Yeah I would be tempted to phrase these consistently with Apple's phrasing, as @tvolkert states.

Other than adding a set of macOS 14 bots as mentioned below. We haven't changed much in terms of our CI config, nor have we dropped support for older versions, so in terms of the way the [Supported deployment platforms](https://docs.flutter.dev/reference/supported-platforms) page is laid out today, I'd say:

* Supported: Ventura (13), Sonoma (14)
* Best efforts versions: Mojave (10.14) to Monterey (12) and Sequoia (15).
* Unsupported: High Sierra (10.13) and earlier.

File: src/_data/platforms.yml [hunk 8]
```diff
@@ -1,50 +1,50 @@
...
   unsupported: '8 and earlier'
 - platform: 'Debian (Linux)'
   chipsets: 'x64, Arm64'
   supported: '11, 12'
-  besteffort: '10'
+  ci-tested: 
   unsupported: '9 and earlier'
 - platform: 'Ubuntu (Linux)'
   chipsets: 'x64, Arm64'
   supported: '20.04 LTS, 22.04 LTS'
```
Comment (cbracken) [inline src/_data/platforms.yml:28]: If this now includes "best effort" then "20.04 to 24.04 LTS"

File: src/_data/platforms.yml (hunk 5 above)
Comment (cbracken) [reply src/_data/platforms.yml:24]: @christopherfujino do we test plain old Debian on the bots at all any more? I thought we'd updated everything to Ubuntu, or have I got this backwards?

Comment (christopherfujino) [reply src/_data/platforms.yml:24]: as a target platform?

Comment (christopherfujino) [reply src/_data/platforms.yml:24]: I guess it doesn't matter, from a quick look at all of our bots, all of our Linux bots look Ubuntu to me.

File: src/_data/platforms.yml (hunk 7 above)
Comment (jmagman) [reply src/_data/platforms.yml:14]: CI tested is macOS Ventura (13) and macOS Sonoma (14).

File: src/_data/platforms.yml (hunk 8 above)
Comment (sfshaza2) [reply src/_data/platforms.yml:28]: No, I don't think so.

File: src/_data/platforms.yml [hunk 9]
```diff
@@ -1,50 +1,50 @@
...
   supported: '21 to 34'
-  besteffort: 'N/A'
+  ci-tested: '21 to 34'
   unsupported: '20 and earlier'
 - platform: 'iOS'
   chipsets: 'Arm64'
-  supported: '17'
-  besteffort: '12 to 16, 18'
+  supported: '12 to 18'
+  ci-tested: '16, 17'
```
Comment (zanderso) [inline src/_data/platforms.yml:9]: @jmagman do we have 16 on CI or only 17?

File: src/_data/platforms.yml [hunk 10]
```diff
@@ -1,50 +1,50 @@
...
   unsupported: '9 and earlier'
 - platform: 'Ubuntu (Linux)'
   chipsets: 'x64, Arm64'
   supported: '20.04 LTS, 22.04 LTS'
-  besteffort: '24.04 LTS'
+  ci-tested: '22.04 LTS'
   unsupported: '23.10 and earlier non-LTS'
 - platform: 'Chrome (Web)'
   chipsets: 'N/A'
   supported: '[Latest 2](https://chromereleases.googleblog.com/search/label/Stable%20updates)'
```
Comment (zanderso) [inline src/_data/platforms.yml:33]: I think this should be `96 until Latest 2`. Same comment for other browsers. @yjbanov to confirm.

File: src/_data/platforms.yml [hunk 11]
```diff
@@ -1,50 +1,50 @@
...
-  besteffort: '96 to latest 2'
+  ci-tested:
```
Comment (zanderso) [inline src/_data/platforms.yml:34]: @yjbanov what browser versions do we have on CI?

File: src/_data/platforms.yml (hunk 9 above)
Comment (jmagman) [reply src/_data/platforms.yml:9]: Only 17.

File: src/_data/platforms.yml (hunk 11 above)
Comment (yjbanov) [reply src/_data/platforms.yml:34]: Engine:
  - Chrome: 119
  - Firefox: 106
  - Safari: depends on the version of macOS 13 (if latest, then it's Safari 18)
Framework:
  - Chrome: 125

File: src/_data/platforms.yml (hunk 10 above)
Comment (yjbanov) [reply src/_data/platforms.yml:33]: Are we changing the structure of the table? What we currently have in https://docs.flutter.dev/reference/supported-platforms LGTM.

Comment (tvolkert) [reply src/_data/platforms.yml:33]: We're getting rid of "best effort" and replacing it with "CI-tested versions" (which of course has different values in it)

File: src/_data/platforms.yml [hunk 12]
```diff
@@ -1,50 +1,50 @@
...
-  supported: '11, 12'
-  besteffort: '10'
+  supported: '10, 11, 12'
+  ci-tested: '11, 12'
   unsupported: '9 and earlier'
 - platform: 'Ubuntu (Linux)'
   chipsets: 'x64, Arm64'
   supported: '20.04 LTS, 22.04 LTS'
-  besteffort: '24.04 LTS'
+  ci-tested: '96 unil Latest 2'
```
Comment (tvolkert) [inline src/_data/platforms.yml:29]: until

File: src/_data/platforms.yml [hunk 13]
```diff
@@ -1,50 +1,50 @@
...
   chipsets: 'N/A'
   supported: '[Latest 2](https://www.mozilla.org/en-US/firefox/releases/)'
-  besteffort: '99 to latest 2'
+  ci-tested: '106'
   unsupported: '98 and earlier'
 - platform: 'Safari (Web)'
   chipsets: 'N/A'
   supported: '[Latest 2](https://developer.apple.com/documentation/safari-release-notes/)'
-  besteffort: '15.6 to latest 2'
+  ci-tested: 'If on latest MacOS 13, then Safari 18'
```
Comment (tvolkert) [inline src/_data/platforms.yml:44]: Just say "18"

File: src/_data/platforms.yml [hunk 14]
```diff
@@ -1,50 +1,50 @@
...
   unsupported: '15.5 and earlier'
 - platform: 'Edge (Web)'
   chipsets: 'N/A'
   supported: '[Latest 2](https://learn.microsoft.com/en-us/deployedge/microsoft-edge-relnote-stable-channel)'
-  besteffort: '96 to latest 2'
+  ci-tested:
```
Comment (tvolkert) [inline src/_data/platforms.yml:49]: This needs a value.  @yjbanov

File: src/_data/platforms.yml (hunk 10 above)
Comment (yjbanov) [reply src/_data/platforms.yml:33]: @zanderso just explained to me how the table is changing. So for web we'll want this:

- Chrome and Edge:
  - Supported: Latest 2 versions
  - CI-tested: 119, 125
- Firefox:
  - Supported: 106 and newer
  - CI-tested: 106
- Safari. I'm still confirming the exact CI version we're using, but we _want_ it to be this:
  - Supported: 16.6 and newer
  - CI-tested: 16.6

**Review by cbracken — APPROVED** [review]
lgtm for macOS, Windows, Linux, and iOS.

**Review by mit-mit — APPROVED** [review]

**Review by zanderso — APPROVED** [review]

**Pull request merged**